### ISSUE TITLE
Derive Component for HashMap field

### ISSUE THREAD

**Issue by kellpossible** [juhaku/utoipa#177]
```rust
#[derive(Component)]
struct Foo {
    field: HashMap<String, u64>,
}
```

results in:

```json
{
  "properties": {
    "field": {
      "type": "object"
    }
  },
  "required": [
    "field"
  ],
  "type": "object"
}
```

It looks like the spec https://swagger.io/specification/#schema-object should support the `additionalProperties` type that can be used to provide a proper schema for the `HashMap`?

```json
{
  "properties": {
    "field": {
      "type": "object",
      "additionalProperties": {
        "type": "integer",
        "format": "uint64",
        "minimum": 0
      }
    }
  }
  "required": [
    "field"
  ],
  "type": "object",
}
```

**Comment by juhaku** [comment]
Nice, this is one of the extra things not yet implemented. But for sure support for additionalProperties could be added as well. It also should not be a super big piece of code to add.

**Comment by juhaku** [comment]
Implemented by #218

**Comment by Ryan-DL** [comment]
@juhaku Has this been pushed into a release? I would like to see this soon 😄
